chainer/optimizer.py

```diff
@@ -518,3 +523,50 @@ def __call__(self, opt):
                 grad = param.grad
                 with cuda.get_device(grad):
                     grad *= rate
+
+
+class GradientNoise(object):
+    """Optimizer hook function for adding gradient noise.
+
+    This hook function simply adds time-dependent annealed Gaussian noise
+    to the gradient at every training step:
+
+    . math::
+
+        g_t \\leftarrow g_t + N(0, \\sigma_t^2)
+
+    where
+
+    . math::
+
+        \\sigma_t^2 = \\frac{\\eta}{(1+t)^\\gamma}
+
+    with :math:`\\eta` selected from {0.01, 0.3, 1.0} and
```

Comment:
We do not restrict the range of `eta` to these three values in this hook. So we need to clear that this choice is done in the original paper.